crates/atuin-daemon/build.rs

```diff
@@ -1,4 +1,17 @@
-fn main() -> Result<(), Box<dyn std::error::Error>> {
-    tonic_build::compile_protos("./proto/history.proto")?;
-    Ok(())
+use std::{env, fs, path::PathBuf};
+
+use protox::prost::Message;
+
+fn main() -> std::io::Result<()> {
+    let file_descriptors = protox::compile(["history.proto"], ["./proto/"]).unwrap();
+
+    let file_descriptor_path = PathBuf::from(env::var_os("OUT_DIR").expect("OUT_DIR not set"))
+        .join("file_descriptor_set.bin");
+    fs::write(&file_descriptor_path, file_descriptors.encode_to_vec()).unwrap();
+
+    tonic_build::configure()
+        .build_server(true)
+        .file_descriptor_set_path(&file_descriptor_path)
+        .skip_protoc_run()
+        .compile(&["root.proto"], &["."])
```

Comment:
```suggestion
        .compile(&["history.proto"], &["."])
```

Reply:
Huh. copypasta. I was wondering why this even worked with the typo, so I dug a bit in the source of prost-build. Turns out when `skip_protoc_run` is set, this string is used for almost nothing except a `cargo:rerun-if-changed` directive. So the build works the same, but the copypasta would lead to the build being cached incorrectly if the protobuf definitions are changed.

By the way, do you prefer incremental commits on a PR or clean history with amend / rebase & force-push?

Reply:
I was also surprised it built ok! That makes sense, thanks for digging into it.

Whichever works best for you. If it's the sort of larger/complex PR where incremental commits tell a better story and are easier to review then it's appreciated, but it's all squashed anyway so not a big deal.